DOC-30 | DOC-33 | Clarifications around ArangoSearch (#1225)
arangodb/docs: 3.10/analyzers.md

[Word Boundaries](https://www.unicode.org/reports/tr29/#Word_Boundaries){:target="_blank"}
in the Unicode Standard Annex #29 about Unicode text segmentation for details.

These tokenizing Analyzers extract tokens, which removes characters like
punctuation and whitespace. An exception is the [`segmentation` Analyzer](#segmentation)
if you select `"graphic"` or `"all"` for the `break` option. They preserve `@`
and `.` characters of email addresses, for instance. There are also exceptions
with both Analyzer types for sequences like numbers, for which decimal and
thousands separators (`.` and `,`) are preserved.

Normalization
-------------

The `norm`, `text`, and `segmentation` Analyzer types allow you to convert the
input text to all lowercase or all uppercase for normalization purposes, namely
case insensitive search. Case folding is not supported. Also see
[Case Mapping](https://unicode-org.github.io/icu/userguide/transforms/casemappings.html){:target="_blank"}
in the ICU documentation.

The `norm` and `text` Analyzer types also allow you to convert characters with
diacritical marks to the base characters. This normalization enables
accent-insensitive search.

> 

Reviewer: Are there plans to describe Stemming and n-grams  here?

Author: Out-of-scope for this PR but I added it to the list of https://arangodb.atlassian.net/browse/DOC-242

  `type` and `properties` attributes
  
{% hint 'info' %}
- You cannot use Analyzers of the types `geopoint` and `geojson` in pipelines.
  These Analyzers require additional postprocessing and can only be applied to
  document fields directly.
- The output data type of an Analyzer needs to be compatible with the input
  data type of the next Analyzer in the chain. For example, the `aql` Analyzer
  has a `returnType` property, and if you set it to `number` or `bool`, the
  subsequent Analyzer in the pipeline needs to support this data type as input.

>   Most Analyzers expect string inputs and are thus incompatible.

Reviewer: ```suggestion
  Most Analyzers expect string inputs and are thus incompatible with such setup.
```
Otherwise that reads like "Just all anaylzers are incompatible". That looks a bit strange. I'd say more - why mention this setting of "aql" analyzer here? It should be described in the corresponding analyzer section with note that setting custom return type will prevent this analyzer from being used in the middle of pipeline.

>   data type of the next Analyzer in the chain. For example, the `aql` Analyzer
>   has a `returnType` property, and if you set it to `number` or `bool`, the
>   subsequent Analyzer in the pipeline needs to support this data type as input.
>   Most Analyzers expect string inputs and are thus incompatible.

Author: ```suggestion
  data type of the next Analyzer in the chain. The `aql` Analyzer, in particular,
  has a `returnType` property, and if you set it to `number` or `bool`, the
  subsequent Analyzer in the pipeline needs to support this data type as input.
  Most Analyzers expect string inputs and are thus incompatible with such a setup.
```

It uses binary comparison to determine if an input token should be discarded.
It checks for exact matches. If the input contains only a substring that

> matches one of the defined stop words, then it is not discarded. Longer inputs

Reviewer: I'd say "stopwords" should be used as it matches what we have in the options. No strong opinion though.

Author: The correct spelling appears to be "stop words", which is what we should use if you refer to the concept, and `stopwords` if it's a code value because that's how the API is.
